File: Java/27. Remove Element.java
/**
 * https://leetcode.com/problems/remove-element/
 */

class Solution {
    // O(n) time complexity
    // O(1) space complexity
    public int removeElement(int[] nums, int val) {
        int count = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != val) {
                nums[count++] = nums[i];
            }
        }
        return count;
    }
}

File: Kotlin/27. Remove Element.kt
/**
 * https://leetcode.com/problems/remove-element/
 */

class Solution {
    // O(n) time complexity
    // O(1) space complexity
    fun removeElement(nums: IntArray, `val`: Int): Int {
        var c = 0
        var len = nums.size
        while (c < len) {
            if (nums[c] != `val`) c++
            else {
                nums[c] = nums[len - 1]
                len--
            }
        }
        return c
    }
}